provider/gateway/rest/router.go

```diff
@@ -396,7 +396,9 @@ func createManifestHandler(_ log.Logger, mclient pmanifest.Client) http.HandlerF
 			return
 		}
 
-		if err := mclient.Submit(req.Context(), requestDeploymentID(req), mani); err != nil {
+		subctx, cancel := context.WithTimeout(req.Context(), 20*time.Second)
```

Comment:
constant up top please.  these kinds of things would ideally be configurable with environment variables.

Reply:
> Some stylistic changes requested, especially around hard-coded durations.
> 
> This all seems unnecessarily complicated (pre-existing cruft), I'll have to take a longer look at it, but we can give this a try.

I think I see a simpler way of handling this, without the goroutine explosion I created. I'll give it a try while I incorporate your other comments.